atomic_reactor/plugins/pre_koji_parent.py

```diff
@@ -226,9 +227,16 @@ def manifest_list_entries_match(self, image, build_id):
             v2_digest = archive['extra']['docker']['digests'][v2_type]
             koji_archives_data[arch] = v2_digest
 
+        # Additional platforms might contain v2 digests of different builds
```

Comment:
I thought @MartinBasti was against mentioning external registries in this upstream repo.  What would you like to have in there? 

```
# Additional platforms might contain v2 digests of different builds
# Identified previously with external non-osbs re-pushed manifest list
```

Reply:
I believe what Martin did not want to see here (and I back him on this) is that we do not want mentions to deployment specific details, such as where/how a specific container registry is accessed.

What I would like to see is a clearer comment on why we are removing certain architectures from the manifest list. It may not be clear for a reader that may be thinking that it was OSBS who created the manifest list at the first place.

This could be a longer comment like: 
```
manifest lists can be manually pushed to the registry to make sure a specific tag
(e.g., latest) is available for all platforms.
In such cases these manifest lists may include images from different koji builds.
We only want to check the digests for the images built in the current parent koji build
```

Reply:
looks good, only  maybe :
"We only want to check the digests for the images built in the current parent koji build" =>
"We only want to check the digests for platforms built in the current koji build"

Reply:
Added check for requested platforms

Reply:
@athos @rcerven  if this is what we wanted I'll update unit tests